api/core/rag/extractor/excel_extractor.py

```diff
@@ -27,10 +28,37 @@ def __init__(
         self._autodetect_encoding = autodetect_encoding
 
     def extract(self) -> list[Document]:
+        """ parse excel file"""
+        if self._file_path.endswith('.xls'):
+            return self._extract4xls()
+        elif self._file_path.endswith('.xlsx'):
+            return self._extract4xlsx()
+
+    def _extract4xls(self) -> list[Document]:
+        wb = xlrd.open_workbook(filename=self._file_path)
+        row_header = None
+        documents = []
+        # loop over all sheets
+        for sheet in wb.sheets():
+            for row_index, row in enumerate(sheet.get_rows(), start=1):
+                if self.is_blank_row(row):
+                    continue
+                if row_header is None:
+                    row_header = row
+                    continue
+                item_arr = []
+                for index, cell in enumerate(row):
+                    txt_value = str(cell.value)
+                    item_arr.append(f'{row_header[index].value}:{txt_value}')
+                item_str = "\n".join(item_arr)
+                document = Document(page_content=item_str, metadata={'source': self._file_path})
+                documents.append(document)
+        return documents
+
+    def _extract4xlsx(self) -> list[Document]:
         """Load from file path using Pandas."""
         data = []
-
-        # 使用 Pandas 读取 Excel 文件的每个工作表
+        # Read each worksheet of an Excel file using Pandas
         xls = pd.ExcelFile(self._file_path)
```

Comment:
`pd.ExcelFile(self._file_path)` can read xlsx and xls, It's not necessary to add functions `_extract4xls` and `_extract4xlsx` ?

Reply:
Yes, we need it. Pandas cannot read media information. We may need to read media information later.